src/platforms/posix/drivers/df_trone_wrapper/df_trone_wrapper.cpp

```diff
@@ -0,0 +1,351 @@
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file df_trone_wrapper.cpp
+ * Driver to access the TROne of the DriverFramework.
+ *
+ * @author Nicolas de Palezieux <ndepal@gmail.com>
+ */
+
+#include <px4_config.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include <unistd.h>
+#include <px4_getopt.h>
+#include <errno.h>
+#include <string>
+
+#include <systemlib/perf_counter.h>
+#include <systemlib/err.h>
+
+#include <drivers/drv_range_finder.h>
+
+#include <uORB/uORB.h>
+#include <uORB/topics/subsystem_info.h>
+#include <uORB/topics/distance_sensor.h>
+
+#include <board_config.h>
+
+#include <trone/TROne.hpp>
+#include <DevMgr.hpp>
+
+
+extern "C" { __EXPORT int df_trone_wrapper_main(int argc, char *argv[]); }
+
+using namespace DriverFramework;
+
+
+class DfTROneWrapper : public TROne
+{
+public:
+	DfTROneWrapper();
+	~DfTROneWrapper();
+
+
+	/**
+	 * Start automatic measurement.
+	 *
+	 * @return 0 on success
+	 */
+	int		start();
+
+	/**
+	 * Stop automatic measurement.
+	 *
+	 * @return 0 on success
+	 */
+	int		stop();
+
+private:
+	int _publish(struct range_sensor_data &data);
+
+	orb_advert_t		_range_topic;
+
+	int			_orb_class_instance;
+
+	// perf_counter_t		_range_sample_perf;
+
+};
+
+DfTROneWrapper::DfTROneWrapper(/*enum Rotation rotation*/) :
+	TROne(TRONE_DEVICE_PATH),
+	_range_topic(nullptr),
+	// _param_update_sub(-1),
+	_orb_class_instance(-1)//,
+	// _range_sample_perf(perf_alloc(PC_ELAPSED, "df_range_read"))
+{
+}
+
+DfTROneWrapper::~DfTROneWrapper()
+{
+	// perf_free(_range_sample_perf);
+}
+
+int DfTROneWrapper::start()
+{
+	struct distance_sensor_s d;
+	_range_topic = orb_advertise_multi(ORB_ID(distance_sensor), &d,
+								     &_orb_class_instance, ORB_PRIO_DEFAULT);
+
+	int ret;
+
+	/* Init device and start sensor. */
+	ret = init();
+	if (ret != 0) {
+		PX4_ERR("TROne init fail: %d", ret);
+		return ret;
+	}
+
+	ret = TROne::start();
+
+	if (ret != 0) {
+		PX4_ERR("TROne start fail: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int DfTROneWrapper::stop()
+{
+	/* Stop sensor. */
+	int ret = TROne::stop();
+
+	if (ret != 0) {
+		PX4_ERR("TROne stop fail: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int DfTROneWrapper::_publish(struct range_sensor_data &data)
+{
+	if (!_range_topic)
```

Comment:
Please always put curly brackets for if/for/while.
Also `make check_format` should catch this.